src/hotspot/share/runtime/objectMonitor.cpp

```diff
@@ -276,7 +276,6 @@ ObjectMonitor::ObjectMonitor(oop object) :
 { }
 
 ObjectMonitor::~ObjectMonitor() {
-  _object.release(_oop_storage);
```

Comment:
Just to be clear, we can't call release here because not all
`delete monitor` calls happen while the JavaThread is in VM.
Some happen while the JavaThread is blocked.

Reply:
Now we use `!_object.is_null()` as a flag to indicate when it is
okay (and safe) to do the release.

Reply:
This is confusing - why is `is_null` being used as a proxy for checking the current thread's state?

Reply:
I add a clarification to my comment above:

Update: s/(and safe)/(and presumed to be safe)/